cmd/docker-proxy/network_proxy_test.go

```diff
@@ -192,33 +204,31 @@ func TestTCP4ProxyHalfClose(t *testing.T) {
 }
 
 func TestTCP6Proxy(t *testing.T) {
-	t.Skip("Need to start CI docker with --ipv6")
 	backend := NewEchoServer(t, "tcp", "[::1]:0", EchoServerOptions{})
 	defer backend.Close()
 	backend.Run()
 	frontendAddr := &net.TCPAddr{IP: net.IPv6loopback, Port: 0}
-	proxy, err := NewProxy(frontendAddr, backend.LocalAddr())
+	listenFd, frontendAddr := tcpListener(t, "tcp6", frontendAddr)
+	proxy, err := NewTCPProxy(listenFd, backend.LocalAddr().(*net.TCPAddr))
 	if err != nil {
 		t.Fatal(err)
 	}
-	testProxy(t, "tcp", proxy, false)
+	testProxyAt(t, "tcp", proxy, frontendAddr.String(), false)
 }
 
 func TestTCPDualStackProxy(t *testing.T) {
```

Comment:
`newProxy` can't create a dual-stack listener - and the daemon doesn't, at the moment.

But, when the socket's passed in, its creation is outside docker-proxy's control, and there's no reason it must v6-only. So, I don't think dropping the test is a good idea.

(The test was skipped before this PR, because it listened on `::1` instead of `::`. So, the socket ended up being IPv6-only and the test failed. But that's fixed now.)